dynamic-security-analysis/src/main/java/com/powsybl/security/dynamic/DynamicSecurityAnalysisProvider.java

```diff
@@ -0,0 +1,149 @@
+ * through {@link com.powsybl.commons.extensions.Extension Extensions}.
+ *
+ * @author Laurent Issertial {@literal <laurent.issertial at rte-france.com>}
+ */
+public interface DynamicSecurityAnalysisProvider extends Versionable, PlatformConfigNamedProvider {
+
+    static List<DynamicSecurityAnalysisProvider> findAll() {
+        return Lists.newArrayList(ServiceLoader.load(DynamicSecurityAnalysisProvider.class, DynamicSecurityAnalysisProvider.class.getClassLoader()));
+    }
+
+    /**
+     * Run an asynchronous single dynamic security analysis job.
+     * <p>
+     * This method should complete with an exception if there is an exception thrown in the security analysis, in order
+     * to be able to collect execution's logs in that case.
+     * The original exception should be wrapped in {@link com.powsybl.computation.ComputationException}, together with
+     * the out and err logs, within a {@link java.util.concurrent.CompletionException}.
+     *
+     * <p>
+     * Example of use:
+     * <pre> {@code
+     * try {
+     *       SecurityAnalysisResult result = dynamicSecurityAnalysis.run(network, variantId, dynamicModelsSupplier, eventModelsSupplier, contingenciesProvider, runParameters).join();
+     *   } catch (CompletionException e) {
+     *       if (e.getCause() instanceof ComputationException) {
+     *           ComputationException computationException = (ComputationException) e.getCause();
+     *           System.out.println("Consume exception...");
+     *           computationException.getOutLogs().forEach((name, content) -> {
+     *               System.out.println("-----" + name + "----");
+     *               System.out.println(content);
+     *           });
+     *           computationException.getErrLogs().forEach((name, content) -> {
+     *               System.out.println("-----" + name + "----");
+     *               System.out.println(content);
+     *           });
+     *       }
+     *       throw e;
+     *   }
+     * }</pre>
+     *
+     * @param network IIDM network on which the security analysis will be performed
+     * @param workingVariantId network variant ID on which the analysis will be performed
+     * @param dynamicModelsSupplier supplies list of dynamic models for the dynamic simulation part
+     * @param eventModelsSupplier supplies list of event models for the dynamic simulation part
+     * @param contingenciesProvider provides list of contingencies
+     * @param runParameters runner parameters
+     * @return a {@link CompletableFuture} on {@link SecurityAnalysisResult} that gathers security factor values
+     */
+    CompletableFuture<SecurityAnalysisReport> run(Network network,
+                                                  String workingVariantId,
+                                                  DynamicModelsSupplier dynamicModelsSupplier,
+                                                  EventModelsSupplier eventModelsSupplier,
```

Comment:
The EventModelsSupplier should be removed, there is no event we want to add outside the contingencies event.